### PR TITLE
ParticleSpawner::step cleanup and rotation fix

### PR DESCRIPTION
**Pull request by SmallJoker**
Removal of duplication code and code style fixes
Fixes rotation issue as described in 8c16f18
Could replace #6484

### CODE REVIEW

File: src/particles.cpp [hunk 1]
```diff
@@ -302,128 +354,33 @@ void ParticleSpawner::step(float dtime, ClientEnvironment* env)
...
-	if (m_spawntime != 0) // Spawner exists for a predefined timespan
-	{
-		for(std::vector<float>::iterator i = m_spawntimes.begin();
-				i != m_spawntimes.end();)
-		{
-			if ((*i) <= m_time && m_amount > 0)
-			{
+	if (m_spawntime != 0) {
+		// Spawner exists for a predefined timespan
+		for (std::vector<float>::iterator i = m_spawntimes.begin();
```
Comment (nerzhul) [inline src/particles.cpp]: please use c++11 styled loop ? :)

File: src/particles.cpp [hunk 2]
```diff
@@ -281,6 +281,58 @@ ParticleSpawner::ParticleSpawner(IGameDef *gamedef, LocalPlayer *player,
...
+
+	if (is_attached) {
+		// Apply attachment yaw
+		vel.rotateXZBy(attached_yaw);
+		acc.rotateXZBy(attached_yaw);
+	}
+
+	float exptime = rand() / (float)RAND_MAX
+			* (m_maxexptime - m_minexptime)
+			+ m_minexptime;
```
Comment (sfan5) [inline src/particles.cpp]: don't we have a macro to do this?

File: src/particles.cpp (hunk 1 above)
Comment (SmallJoker) [reply src/particles.cpp]: Can't, as we erase entries from the vector while iterating.

File: src/particles.cpp (hunk 2 above)
Comment (SmallJoker) [reply src/particles.cpp]: `linearInterpolation` in `noise.cpp` does that but I couldn't find a macro or static function.

**Pull request merged**